aips/aip-x.md

```diff
@@ -0,0 +1,707 @@
+
+#### Suggested Features
+
+The features below will be added to the wallet as suggested features. Features will be optional to maintain compatibility with existing wallets.
+
+`aptos:signIn` method to establish a connection and to sign a _Sign in with Aptos_ authentication message.
+
+```ts
+// `input: AptosSignInInput` - inputs with fields to construct an authentication message
+signIn(input: AptosSignInInput): Promise<UserResponse<AptosSignInOutput>>
+```
+
+### Wallet Implementer
+
+To ensure the safety of our users, we want to take a conservative approach when allowing users to sign potentially harmful messages.
+
+#### Bound Fields Verification
+
+The following fields MUST be verified to prevent phishing attacks: `domain` , `address` , `uri` , `version` , and `chainId` . If the fields do not match the signing request, errors must be displayed to the users detailing them out. It is suggested to add additional steps to the approval process, such as an opt-in checkmark field, that confirms the user’s intent if errors are present.
+
+A wallet may also decide to prevent the approval of invalid signing requests.
+
+## Reference Implementation
+
+### Interfaces
+
+#### @aptos-labs/siwa
+
+A new package named `@aptos-labs/siwa` will be created to house the utilities for the _Sign in with Aptos_ standard. These utilities will be responsible for creating messages, verifying messages, and deserializing public key and/or signatures according.
+
+Below are the primary functions that that will be provided as part of the `@aptos-labs/siwa` utilities:
+
+```ts
+/**
+ * Create a SignIn message text from the input following the ABNF format defined in the Sign in with Aptos
+ * specifications.
+ *
+ * @param input The input to create the SignIn message text from.
+ *
+ * @returns The SignIn message text.
+ */
+function createSignInMessageText(input: AptosSignInMessage["fields"]): string;
+
+/**
+ * Verifies an input SignIn message against expected fields.
+ *
+ * @param input The input to verify the message against.
+ * @param expected The expected message to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignInMessage(
+  input: AptosSignInInput,
+  expected: AptosSignInMessage | AptosSignInMessage["fields"] | string
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Verifies outputs from a `signIn` method response against input fields.
+ *
+ * @param input The input to verify the output against.
+ * @param output The output to verify against the input.
+ *
+ * @returns The verification result.
+ */
+function verifySignIn(
+  input: AptosSignInInput,
+  output: {
+    publicKey: PublicKey;
+    signature: Signature;
+    message: string;
+  }
+): VerificationResult<AptosSignInMessage["fields"]>;
+
+/**
+ * Generate a signing message using the Sign in with Aptos signing algorithm.
```

Comment:
Is a "signing message" different than a "[Aptos]SignInMessage"?

If so, did the AIPs text properly introduce both & differentiate between them? I don't think "signing message" has been introduced.